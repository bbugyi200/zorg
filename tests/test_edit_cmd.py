"""Tests the zorg project's 'edit' CLI command."""

from __future__ import annotations

from pathlib import Path

from syrupy.assertion import SnapshotAssertion as Snapshot

from . import common as c


def test_edit_day_logs(
    main: c.MainType, tmp_path: Path, snapshot: Snapshot
) -> None:
    """Test that we properly generate test day, habit, and done logs."""
    zettel_dir = tmp_path / "org"
    kwargs = {}
    for key, stem in [
        ("day_log_template", "day_log_tmpl"),
        ("habit_log_template", "habit_log_tmpl"),
        ("done_log_template", "done_log_tmpl"),
    ]:
        template_path: Path = zettel_dir / f"{stem}.zo"
        template_path.parent.mkdir(parents=True, exist_ok=True)
        test_data_template_path = Path(__file__).parent / Path(
            f"data/{stem}.zo"
        )
        template_path.write_text(test_data_template_path.read_text())
        kwargs[key] = template_path.name

    day_log = f"{c.YYYY}/{c.YYYY}{c.MM}{c.DD}.zo"
    habit_log = f"{c.YYYY}/{c.YYYY}{c.MM}{c.YEST_DD}_habit.zo"
    done_log = f"{c.YYYY}/{c.YYYY}{c.MM}{c.DD}_done.zo"
    re_date_group = "(?P<date>[0-9]{4}[01][0-9][0-3][0-9])"
    exit_code = main(
        "--dir",
        str(zettel_dir),
        "edit",
        day_log,
        habit_log,
        done_log,
        template_pattern_map={
            f"^{re_date_group}_habit$": "habit_log_tmpl.zo",
            f"^{re_date_group}_done$": "done_log_tmpl.zo",
            f"^{re_date_group}$": "day_log_tmpl.zo",
        },
        **kwargs,
    )

    day_log_path = zettel_dir / day_log
    habit_log_path = zettel_dir / habit_log
    done_log_path = zettel_dir / done_log

    assert exit_code == 0
    assert day_log_path.read_text() == snapshot
    assert habit_log_path.read_text() == snapshot
    assert done_log_path.read_text() == snapshot
