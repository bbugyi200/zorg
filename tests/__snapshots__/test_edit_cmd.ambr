# serializer version: 1
# name: test_edit_day_logs[YAMLConfigFile]
  '''
  # 2000-01-03.Mon
  #
  # @ = [[day_log_tmpl]]
  # < = [[20000102.zo]]
  # > = [[20000104.zo]]
  #
  # H = [[20000103_habit.zo]]
  # X = [[20000103_done.zo]]
  
  o #gtd pomodoro
    * Process yesterday's bujo log (e.g. close events, add missing pomodoros, rollover goals)
    * Copy any important gchat convos from yesterday into chat/*.txt
    * Google Calendar (bbugyi@google + bryanbugyi34@gmail)
    * Check today's tickler file: [[tick_3]]
    * Fill out yesterday's habit tracker: [[20000102_habit.zo]]
    * go/g3co
    * Review '@A' todos (are any >=7d old? which should you do today?)
    * Set 2-5 goals / "highlights" for today
    * Add [[act]] file links to this day log
    * <50 emails in bryanbugyi34@gmail
    * <100 emails in bbugyi@google
    * inbox: Google Keep
    * inbox: ~/org/box.zo
    * Review ~/org diffs from yesterday and commit them to git
  
  ##### Goals #####
  
  ##### Events #####
  
  ##### Pomodoros #####
  '''
# ---
# name: test_edit_day_logs[YAMLConfigFile].1
  '''
  # Habits | 2000-01-02.Sun
  #
  # @ = [[habit_log_tmpl]]
  # < = [[20000101_habit.zo]]
  # > = [[20000103_habit.zo]]
  #
  # 0 = [[20000102.zo]]
  
  meds: ?
  run: ?
  time_of_1st_pomodoro: ????
  w: ?
  wfo: ?
  ----- CL -----
  R0 bounces: ?
  RN bounces: ?
  submits: ?
  ----- pm routine -----
  picked clothes: ?
  brush teeth: ?
  rubber bands: ?
  ----- pomodoros -----
  all: ?
  overflow: ?
  work: ?
  
  ##### expenses #####
  
  total:
  '''
# ---
# name: test_edit_day_logs[YAMLConfigFile].2
  '''
  # 2000-01-03.Mon | Done Tasks / Projects
  #
  # @ = [[done_log_tmpl]]
  # < = [[20000102_done.zo]]
  # > = [[20000104_done.zo]]
  #
  # 0 = [[20000103.zo]]
  
  
  '''
# ---
